src/UWP/ArcGISRuntime.UWP.Viewer/Samples/Network Analysis/ClosestFacility/ClosestFacility.xaml.cs

```diff
@@ -0,0 +1,171 @@
+                new Facility(new MapPoint(-1.3042129900625112E7, 3860127.9479775648, _spatialReference)),
+                new Facility(new MapPoint(-1.3042193400557665E7, 3862448.873041752, _spatialReference)),
+                new Facility(new MapPoint(-1.3046882875518233E7, 3862704.9896770366, _spatialReference)),
+                new Facility(new MapPoint(-1.3040539754780494E7, 3862924.5938606677, _spatialReference)),
+                new Facility(new MapPoint(-1.3042571225655518E7, 3858981.773018156, _spatialReference)),
+                new Facility(new MapPoint(-1.3039784633928463E7, 3856692.5980474586, _spatialReference)),
+                new Facility(new MapPoint(-1.3049023883956768E7, 3861993.789732541, _spatialReference))
+                };
+
+            // Create a symbol for displaying facilities.
+            _facilitySymbol = new PictureMarkerSymbol(_pictureMarkerUri);
+            _facilitySymbol.Height = 30;
+            _facilitySymbol.Width = 30;
+
+            // Incident symbol
+            _incidentSymbol = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle.Cross, System.Drawing.Color.FromArgb(255, 0, 0, 0), 20);
+
+            // Route to hospital symbol
+            _routeSymbol = new SimpleLineSymbol(SimpleLineSymbolStyle.Solid, System.Drawing.Color.FromArgb(255, 0, 0, 255), 2.5f);
+
+            // Create Graphics Overlays for incidents and facilities.
+            _incidentGraphicsOverlay = new GraphicsOverlay();
+            _facilityGraphicsOverlay = new GraphicsOverlay();
+
+            // Create a graphic and add to graphics overlay for each facility.
+            foreach (Facility facility in _facilities)
+            {
+                _facilityGraphicsOverlay.Graphics.Add(new Graphic(facility.Geometry, _facilitySymbol));
+            }
+
+            // Add each graphics overlay to MyMapView.
+            MyMapView.GraphicsOverlays.Add(_facilityGraphicsOverlay);
+            MyMapView.GraphicsOverlays.Add(_incidentGraphicsOverlay);
+
+            // Link the action of tapping on the map with the MyMapView_GeoViewTapped method.
+            MyMapView.GeoViewTapped += MyMapView_GeoViewTappedAsync;            
+
+        }
+
+        private void MyMapView_GeoViewTappedAsync(object sender, Esri.ArcGISRuntime.UI.Controls.GeoViewInputEventArgs e)
+        {
+            // Clear any prior incident and routes from the graphics.
+            _incidentGraphicsOverlay.Graphics.Clear();
+
+            // Create a MapPoint where the user clicked.
+            _incidentPoint = MyMapView.ScreenToLocation(e.Position);
+            
+            // Populate the facility parameters than solve using the task.
+            populateParametersAndSolveRouteAsync();
+
+        }
+
+        private async void populateParametersAndSolveRouteAsync()
+        {
+            // Set facilities and incident in parameters.
+            _closestFacilityParameters = _task.CreateDefaultParametersAsync().Result;
```

Comment:
I recommend switching to `await _task.CreateDefaultParametersAsync();`. Async/await stuff is tricky. We can talk offline about some resources for learning more. 

For more detail about this case specifically, see [this StackOverflow answer](https://stackoverflow.com/questions/24623120/await-on-a-completed-task-same-as-task-result#24657079).